src/watertap_contrib/seto/unit_models/surrogate/med_tvc_surrogate.py

```diff
@@ -0,0 +1,1394 @@
+        self.add_port(name="brine", block=self.brine_props)
+        self.add_port(name="steam", block=self.heating_steam_props)
+        self.add_port(name="motive", block=self.motive_steam_props)
+
+        """
+        Mass balances
+        """
+        # Distillate flow rate calculation
+        @self.Constraint(doc="Distallate volumetric flow rate")
+        def eq_dist_vol_flow(b):
+            return (
+                b.distillate_props[0].flow_vol_phase["Liq"]
+                == b.feed_props[0].flow_vol_phase["Liq"]
+                * b.recovery_vol_phase[0, "Liq"]
+            )
+
+        # Brine flow rate calculation
+        @self.Constraint(doc="Brine volume flow rate")
+        def eq_brine_vol_flow(b):
+            return (
+                b.brine_props[0].flow_vol_phase["Liq"]
+                == b.feed_props[0].flow_vol_phase["Liq"]
+                - b.distillate_props[0].flow_vol_phase["Liq"]
+            )
+
+        """
+        Isobaric
+        """
+
+        @self.Constraint(doc="Isobaric")
+        def eq_feed_to_distillate_isobaric(b):
+            return b.feed_props[0].pressure == b.distillate_props[0].pressure
+
+        @self.Constraint(doc="Isobaric")
+        def eq_feed_to_brine_isobaric(b):
+            return b.feed_props[0].pressure == b.brine_props[0].pressure
+
+        @self.Constraint(doc="Isobaric")
+        def eq_feed_to_cooling_isobaric(b):
+            return b.feed_props[0].pressure == b.cooling_out_props[0].pressure
+
+        """
+        Add Vars for model outputs
+        """
+        self.thermal_power_requirement = Var(
+            initialize=5000,
+            bounds=(0, None),
+            units=pyunits.kW,
+            doc="Thermal power requirement (kW)",
+        )
+
+        self.specific_area = Var(
+            initialize=2,
+            bounds=(0, None),
+            units=pyunits.m**2 / (pyunits.m**3 / pyunits.d),
+            doc="Specific area (m2/m3/day))",
+        )
+
+        self.specific_energy_consumption_thermal = Var(
+            initialize=65,
+            bounds=(0, None),
+            units=pyunits.kWh / pyunits.m**3,
+            doc="Specific thermal power consumption (kWh/m3)",
+        )
+
+        self.gain_output_ratio = Var(
+            initialize=10,
+            bounds=(0, None),
+            units=pyunits.kg / pyunits.kg,
+            doc="Gained output ratio (kg of distillate water per kg of heating steam",
+        )
+
+        """
+        Add Vars for intermediate model variables
+        """
+        self.feed_cool_mass_flow = Var(
+            initialize=1000,
+            bounds=(0, None),
+            units=pyunits.kg / pyunits.s,
+            doc="Feed and cooling water mass flow rate (kg/s)",
+        )
+
+        self.feed_cool_vol_flow = Var(
+            initialize=1000,
+            bounds=(0, None),
+            units=pyunits.m**3 / pyunits.h,
+            doc="Feed and cooling water volume flow rate (m3/h)",
+        )
```

Comment:
Unsure why you need these. They should be on the property model as `flow_vol_phase`

Reply:
Ah, perhaps because you didn't feel the need to build a state block for inlet cooling water. Is that right? If so, would be good to add that to the comment above.

Reply:
I these are combined feed and cooling flow? A similar approach is used in LT-MED.

Reply:
Yes, it is just used to calculate the flow rate of the cooling water, and a comment is added above.

Reply:
> > @Zhuoran29 my initial guess is that the infeasible utilities were moved in the latest WaterTAP version on main. Try changing to watertap.core.util.model_diagnostics.infeasible
> 
> @adam-a-a @Zhuoran29 yes that is correct. Ben recently moved them so the import needs to change. One of the pitfalls of pointing at watertap/main.

Thanks, problem solved.

Reply:
Yes, it is just used to calculate the flow rate of the cooling water, and a comment is added above.